lib/src/interable_extensions.dart

```diff
@@ -5,12 +5,15 @@ import 'dart:math' as math;
 
 final _unorderedEquality = <Type, UnorderedIterableEquality>{};
 
-/// Function, that returns `true` if element is pass test.
+/// Function that returns `true` if element passes test.
 typedef TestPredicate<E> = bool Function(E element);
 
-/// Function, that value [T] for tha elemet.
+/// Function that gets value [T] for that element.
 typedef GetValue<E, T> = T Function(E element);
 
+/// Function that returns value [T] for the element and index
+typedef MapIndexedValue<E, T> = T Function(E element, int index);
```

Comment:
@greymag I did the index second to keep it more compatible with `Iterable.map`.

For instance, if you define a method as follows, it can be used with both `map` and `mapIndex`:
```dart
int compatibleCallback(int value, [int? index])
```

When used with `map`, the value of `index` will be `null`. It will be populated when used with `mapIndex`.

However, it's your library, so if you would like me to change it then I will.

Reply:
(this also matches the javascript method for where the index is passed)

Reply:
@greymag do you want me to make this change, or just leave it as is?

Reply:
No, that's OK. Your explanation was reasonable and clear, thank you!